File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 3/31/2026, 9:41:24 PM
#define faster \
ios::sync_with_stdio(false);\
cin.tie(nullptr);\
cout.tie(nullptr);
class Solution {
public:
    int maxArea(vector<int>& height) {
        faster
        int l = 0;
        int r = height.size()-1;
        int maxx = 0,curr=0;
        while(l<r){
            curr = min(height[l], height[r])*(r-l);
            maxx = max(curr, maxx);
            if (height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return maxx;
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 3/31/2026, 9:41:19 PM
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxx = 0
        l, h = 0,len(height)-1
        while l<=h:
            ar = min(height[l], height[h])*(h-l)
            maxx = max(ar, maxx)
            if height[l]<=height[h]:
                l+=1
            else:
                h-=1
        return maxx